libbeat/otelcommon/providers/fbprovider/fbprovider.go

```diff
@@ -0,0 +1,83 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package fbprovider
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap"
+
+	"github.com/elastic/beats/v7/libbeat/cfgfile"
+)
+
+const schemeName = "fb"
+
+type provider struct{}
+
+func NewFactory() confmap.ProviderFactory {
+	return confmap.NewProviderFactory(newProvider)
+}
+
+func newProvider(confmap.ProviderSettings) confmap.Provider {
+	return &provider{}
+}
+
+func (fmp *provider) Retrieve(_ context.Context, uri string, _ confmap.WatcherFunc) (*confmap.Retrieved, error) {
+	if !strings.HasPrefix(uri, schemeName+":") {
```

Comment:
Do we really need the scheme?  Inside the config it says if it is metricbeat, filebeat, etc.

Reply:
yes because it is required for a provider to have a `scheme` associated with it https://github.com/open-telemetry/opentelemetry-collector/tree/main/confmap#provider

An empty scheme name/or no scheme will throw an error
https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/provider.go#L76